src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RlScanStepRunner.java

```diff
@@ -0,0 +1,151 @@
+import com.synopsys.integration.detect.lifecycle.run.operation.OperationRunner;
+import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioHeaderUtil;
+import com.synopsys.integration.detect.workflow.codelocation.CodeLocationNameManager;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.HttpUrl;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.NameVersion;
+
+public class RlScanStepRunner {
+    
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final NameVersion projectNameVersion;
+    private final String projectGroupName;
+    private UUID scanId;
+    private final OperationRunner operationRunner;
+    private final File rlRunDirectory;
+    private final BlackDuckRunData blackDuckRunData;
+    private String codeLocationName;
+    private static final String STORAGE_UPLOAD_ENDPOINT = "/api/storage/rldata/";
+    private static final String STORAGE_RL_CONTENT_TYPE = "application/vnd.blackducksoftware.rl-data-1+octet-stream";
+    private static final BlackDuckVersion MIN_BLACK_DUCK_VERSION = new BlackDuckVersion(2024, 4, 0);
+    
+    public RlScanStepRunner(OperationRunner operationRunner, BlackDuckRunData blackDuckRunData, NameVersion projectNameVersion) {
+        this.operationRunner = operationRunner;
+        this.blackDuckRunData = blackDuckRunData;
+        this.projectNameVersion = projectNameVersion;
+        
+        rlRunDirectory = operationRunner.getDirectoryManager().getReversingLabsOutputDirectory();
+        projectGroupName = operationRunner.calculateProjectGroupOptions().getProjectGroup();
+    }
+
+    public Optional<UUID> invokeRlWorkflow() {
+        try {
+            logger.debug("Determining if configuration is valid to run a ReversingLabs scan.");
+            if (!isReversingLabsEligible()) {
+                logger.info("No detect.rl.scan.file.path property was provided. Skipping ReversingLabs scan.");
+                return Optional.ofNullable(scanId);
+            }
+            if (!isBlackDuckVersionValid()) {
+                String minBlackDuckVersion = String.join(".", 
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getMajor()),
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getMinor()),
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getPatch()));
+                throw new IntegrationException("ReversingLabs scan is only supported with BlackDuck version "
+                        + minBlackDuckVersion + " or greater. ReversingLabs scan could not be run.");
+            }
+
+            codeLocationName = createCodeLocationName();
+            File fileToUpload = new File(operationRunner.getRlScanFilePath().get());
+            
+            // Generate BDIO header and obtain scanID
+            initiateScan(fileToUpload.length());
```

Comment:
Can have an NPE check here (but won't be necessary though if the file-resolved condition is checked early)

Reply:
I think I'll go with the above and leave this line simple.